src/arguments/CoreMember.ts

```diff
@@ -0,0 +1,26 @@
+import type { PieceContext } from '@sapphire/pieces';
+import { Constants, DiscordAPIError, GuildMember } from 'discord.js';
+import { Argument, ArgumentContext, AsyncArgumentResult } from '../lib/structures/Argument';
+
+export class CoreArgument extends Argument<GuildMember> {
+	public constructor(context: PieceContext) {
+		super(context, { name: 'member' });
+	}
+
+	public async run(argument: string, context: ArgumentContext): AsyncArgumentResult<GuildMember> {
+		const { guild } = context.message;
+		if (!guild) {
+			return this.error(argument, 'ArgumentMemberMissingGuild', 'The argument must be run on a guild.');
+		}
+
+		try {
+			return this.ok(await guild.members.fetch(argument));
+		} catch (error) {
+			if (error instanceof DiscordAPIError && error.code === Constants.APIErrors.UNKNOWN_MEMBER) {
```

Comment:
Maybe make Vlad's discord-api-types a dependency and use that instead?

Reply:
Not really necessary, this works for the time being.

Reply:
Using the dependency would make it.. less dependent on discord.js 👀

Reply:
Yeah but we're using `this.client`, which already gives it some dependency to discord.js

Also the fact that SapphireClient extends discord.js's Client.